### DISCUSSION TITLE
Pharmacophore feature definition

### DISCUSSION THREAD

**Discussion by Mirrty** [rdkit/rdkit#4609] [Q&A]
I would like to ask weather my pharmacophore feature assignment is correct:

```
smi='C=CC(=O)N1CCC=C(C(=O)N2CCC(N3CCCC3)CC2)C1'
featFactory = ChemicalFeatures.BuildFeatureFactory('RDKit/Data/BaseFeatures.fdef')
for i in featFactory.GetFeaturesForMol(mymol):
    print (i.GetFamily(),i.GetType(), i.GetAtomIds())
```
```
Donor SingleAtomDonor (2,)
Donor SingleAtomDonor (11,)
Donor SingleAtomDonor (15,)
Acceptor SingleAtomAcceptor (0,)
Acceptor SingleAtomAcceptor (16,)
PosIonizable BasicGroup (15,)
Hydrophobe ThreeWayAttach (3,)
Hydrophobe ChainTwoWayAttach (17,)
```

Atoms 2 and 11 are assign to 'Donor' family, whereas there are no hydrogens attached neither those atoms will be protonated?

![mymol](https://user-images.githubusercontent.com/43479139/137361370-f28d27f4-2d6d-4e24-bfb4-bad25e615caa.png)
Thanks

**Comment by greglandrum** [comment]
The scheme that the default RDKit feature definitions use assumes that molecules are neutral (i.e. non-protonated) and that they should identify protonatable sites and label them as donors. That's what you're seeing here.

Unfortunately the default feature definitions are not great, so they will flag features which are not actually likely to be protonated more often than I like.

So the answer to your question: yes, those sites should be donors based on the default feature definitions, but it could well be that the default feature definitions are wrong in this case.
